Core/src/Geometry/CuboidVolumeStack.cpp

```diff
@@ -0,0 +1,893 @@
+    } else {
+      ACTS_VERBOSE(" -> No overlap");
+    }
+
+    if (std::abs(a.max(m_dir) - b.min(m_dir)) < tolerance) {
+      ACTS_VERBOSE("No gap between volumes, no attachment needed");
+    } else {
+      double gapWidth = b.min(m_dir) - a.max(m_dir);
+      ACTS_VERBOSE("Gap width: " << gapWidth);
+
+      ACTS_VERBOSE("Synchronizing bounds in "
+                   << axisDirectionName(m_dir)
+                   << " with strategy: " << strategy);
+      switch (strategy) {
+        case AttachmentStrategy::Midpoint: {
+          ACTS_VERBOSE(" -> Strategy: Expand both volumes to midpoint");
+
+          double aMidNew = (a.min(m_dir) + a.max(m_dir)) / 2.0 + gapWidth / 4.0;
+          double aHlNew = a.halfLength(m_dir) + gapWidth / 4.0;
+          ACTS_VERBOSE("  - New halflength for first volume: " << aHlNew);
+          ACTS_VERBOSE("  - New bounds for first volume: ["
+                       << (aMidNew - aHlNew) << " <- " << aMidNew << " -> "
+                       << (aMidNew + aHlNew) << "]");
+
+          assert(std::abs(a.min(m_dir) - (aMidNew - aHlNew)) < 1e-9 &&
+                 "Volume shrunk");
+          assert(aHlNew >= a.halfLength(m_dir) && "Volume shrunk");
+
+          double bMidNew = (b.min(m_dir) + b.max(m_dir)) / 2.0 - gapWidth / 4.0;
+          double bHlNew = b.halfLength(m_dir) + gapWidth / 4.0;
+          ACTS_VERBOSE("  - New halflength for second volume: " << bHlNew);
+          ACTS_VERBOSE("  - New bounds for second volume: ["
+                       << (bMidNew - bHlNew) << " <- " << bMidNew << " -> "
+                       << (bMidNew + bHlNew) << "]");
+
+          assert(bHlNew >= b.halfLength(m_dir) && "Volume shrunk");
+          assert(std::abs(b.max(m_dir) - (bMidNew + bHlNew)) < 1e-9 &&
+                 "Volume shrunk");
+
+          Translation3 translationA(Vector3::Unit(dirIdx) * aMidNew);
+          a.setLocalTransform(Transform3{translationA}, m_groupTransform);
+          a.updatedBounds->set(dirBoundIdx, aHlNew);
+
+          Translation3 translationB(Vector3::Unit(dirIdx) * bMidNew);
+          b.setLocalTransform(Transform3{translationB}, m_groupTransform);
+          b.updatedBounds->set(dirBoundIdx, bHlNew);
+
+          break;
+        }
+        case AttachmentStrategy::First: {
+          ACTS_VERBOSE(" -> Strategy: Expand first volume");
+          double aMidNew = (a.min(m_dir) + b.min(m_dir)) / 2.0;
+          double aHlNew = (b.min(m_dir) - a.min(m_dir)) / 2.0;
+          ACTS_VERBOSE("  - Gap width: " << gapWidth);
+          ACTS_VERBOSE("  - New bounds for first volume: ["
+                       << (aMidNew - aHlNew) << " <- " << aMidNew << " -> "
+                       << (aMidNew + aHlNew) << "]");
+
+          assert(std::abs(a.min(m_dir) - (aMidNew - aHlNew)) < 1e-9 &&
+                 "Volume shrunk");
+          assert(aHlNew >= a.halfLength(m_dir) && "Volume shrunk");
+
+          Translation3 translationA(Vector3::Unit(dirIdx) * aMidNew);
```

Comment:
This should only be done if `dirIdx` is ensured to be `0, 1, 2` explicitly. You can do this, if at the entry boundary this index is derived in a safe way.